src/Translator/TranslatableMessageInterface.php

```diff
@@ -0,0 +1,46 @@
+<?php
+/**
+ * Zend Framework (http://framework.zend.com/)
+ *
+ * @link      http://github.com/zendframework/zf2 for the canonical source repository
+ * @copyright Copyright (c) 2005-2015 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license   http://framework.zend.com/license/new-bsd New BSD License
+ */
+
+namespace Zend\Validator\Translator;
+
+/**
+ * This interface provide methods for extract the translatable parts of a message.
+ */
+interface TranslatableMessageInterface
+{
+    /**
+     * Returns the message to translate.
+     *
+     * The message may include placeholders for message variable interpolation.
+     *
+     * @return string
+     */
+    public function getMessageTemplate();
+
+    /**
+     * Returns the variables to be replaced in the message template placeholders.
+     *
+     * @return array
+     */
+    public function getMessageVariables();
+
+    /**
+     * Returns the translation domain namespace.
+     *
+     * @return string
+     */
+    public function getTranslationDomain();
+
+    /**
+     * Returns the message without translation and with the placeholders replaced by the variables.
```

Comment:
Ok, I understand why translation shouldn't happen inside.

However, from a User-PoV the naming then is really confusing. Essentially all I get is a message. I'm able to translate any message by other means, so personally I don't quite see the need to call this `TranslatableMessageInterface` then. Furthermore the messages would be returned from the validator. So the validator needs to know the textdomain as well so it could pass it into the message object. I feel that this is passing along metadata that may not even be needed.

So This would bring us to what @weierophinney proposed (the way I understand it at least): skip translation.

```
foreach ($messages as $msg) {
    $translator->translate((string) $msg);
}
```

Which I feel is perfectly fine. And if you need higher level translation you would probably do something along the lines of

```
foreach ($messages as $msg) {
    $translator->translate($msg->getMessageTemplate(), $msg->getMessageVariables());
}
```

From the user point of view (and the ones of people who write docs), i think this is a pretty good syntax. But with this said my question:
Why the need for anything translation related in those messages ?